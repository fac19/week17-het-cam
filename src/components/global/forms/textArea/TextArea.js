import React from "react";
import { TextAreaLabel, TextAreaBox } from "./TextArea.style";

const TextArea = (props) => {
  return (
    <TextAreaLabel>
      {props.label}
      <TextAreaBox
        name={props.name}
        placeholder="Please type your answer here..."
        onChange={props.onChange}
        required
      />
    </TextAreaLabel>
  );
};

export default TextArea;
