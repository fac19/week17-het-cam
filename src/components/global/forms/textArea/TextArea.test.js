import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import TextArea from "./TextArea";

test("TextArea component renders correctly", () => {
  render(<TextArea label={"Task Two"} />);
  screen.getByText("Task Two");
});

test("TextArea component is styled", () => {
  const tree = renderer.create(<TextArea />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("display", "flex");
});
