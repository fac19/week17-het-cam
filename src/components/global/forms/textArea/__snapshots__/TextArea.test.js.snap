// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea component is styled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  min-height: 5rem;
  resize: none;
}

<label
  className="c0"
>
  <textarea
    className="c1"
    placeholder="Please type your answer here..."
    required={true}
  />
</label>
`;
